plugins/obs-ffmpeg/obs-ffmpeg-mux.c

```diff
@@ -465,10 +358,31 @@ static int deactivate(struct ffmpeg_muxer *stream, int code)
 	}
 
 	os_atomic_set_bool(&stream->stopping, false);
+
+	if (stream->is_hls) {
```

Comment:
I think we better write a different deactivate function for hls. I think we can't just destroy the pipe like this because the write thread is still running and it may write to the pipe.

Reply:
switched the order for stream->is_hls